### ISSUE TITLE
Latest versions don't work with multiple ThrowsExactly in Assert.Multiple

### ISSUE THREAD

**Issue by Patrick8639** [thomhurst/TUnit#1345]
To reproduce:

```cs
  [Test]
  public async Task TestException ()
  {
    using (Assert.Multiple ())
    {
      await Assert.That (
        static () => throw new InvalidOperationException ()
      ).ThrowsExactly <InvalidOperationException> ();
      await Assert.That (
        static () => throw new InvalidOperationException ()
      ).ThrowsExactly <InvalidOperationException> ();
    }
  }
```

In TUnit 0.4.49, the test is successful.

In later versions, the following error is thrown:
>One or more errors occurred. ((This exception may or may not have been caught) System.InvalidOperationException: Hello World) ((This exception may or ...

>One or more errors occurred. ((This exception may or may not have been caught) System.InvalidOperationException: Hello World) >((This exception may or may not have been caught) System.InvalidOperationException: Hello World)
>   at TUnit.Assertions.AssertionScope.Dispose()
   at TestException() in xxx
   at TestException()
   at TUnit.Core.AsyncConvert.Convert(Func`1 action)
   at TUnit.Core.DiscoveredTest`1.ExecuteTest(CancellationToken cancellationToken)
   at TUnit.Engine.Helpers.Timings.Record(String name, TestContext context, Func`1 action)
   at TUnit.Engine.Services.TestInvoker.Invoke(DiscoveredTest discoveredTest, CancellationToken cancellationToken, List`1 cleanupExceptions)
   at TUnit.Engine.Services.TestInvoker.Invoke(DiscoveredTest discoveredTest, CancellationToken cancellationToken, List`1 cleanupExceptions)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteTestMethodWithTimeout(DiscoveredTest discoveredTest, CancellationToken cancellationToken, List`1 cleanupExceptions)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteWithCancellationTokens(DiscoveredTest discoveredTest, List`1 cleanupExceptions)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteWithRetries(DiscoveredTest discoveredTest, List`1 cleanupExceptions)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteWithRetries(DiscoveredTest discoveredTest, List`1 cleanupExceptions)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteTestInternalAsync(DiscoveredTest test, ITestExecutionFilter filter, ExecuteRequestContext context, Boolean isStartedAsDependencyForAnotherTest)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteTestInternalAsync(DiscoveredTest test, ITestExecutionFilter filter, ExecuteRequestContext context, Boolean isStartedAsDependencyForAnotherTest)
   at TUnit.Engine.Services.SingleTestExecutor.ExecuteTestInternalAsync(DiscoveredTest test, ITestExecutionFilter filter, ExecuteRequestContext context, Boolean isStartedAsDependencyForAnotherTest)
   at TUnit.Engine.Services.TestsExecutor.ProcessTest(DiscoveredTest test, ITestExecutionFilter filter, ExecuteRequestContext context, CancellationToken cancellationToken)

**Comment by thomhurst** [comment]
Thanks for the report! Should be fixed in v0.4.71

**Comment by Patrick8639** [comment]
@thomhurst Thank you Tom, it's working now.
